File: Stripe/STPThreeDSSelectionCustomization.m
//
//  STPThreeDSSelectionCustomization.m
//  StripeiOS
//
//  Created by Yuki Tokuhiro on 6/18/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

#import "STPThreeDSSelectionCustomization.h"
#import "STPThreeDSCustomization+Private.h"

#import <Stripe/STDSSelectionCustomization.h>

@implementation STPThreeDSSelectionCustomization

+ (instancetype)defaultSettings {
    return [STPThreeDSSelectionCustomization new];
}

- (instancetype)init {
    self = [super init];
    if (self) {
        _selectionCustomization = [STDSSelectionCustomization defaultSettings];
    }
    return self;
}

- (UIColor *)primarySelectedColor {
    return self.selectionCustomization.primarySelectedColor;
}

- (void)setPrimarySelectedColor:(UIColor *)primarySelectedColor {
    self.selectionCustomization.primarySelectedColor = primarySelectedColor;
}

- (UIColor *)secondarySelectedColor {
    return self.selectionCustomization.secondarySelectedColor;
}

- (void)setSecondarySelectedColor:(UIColor *)secondarySelectedColor {
    self.selectionCustomization.secondarySelectedColor = secondarySelectedColor;
}

- (UIColor *)unselectedBackgroundColor {
    return self.selectionCustomization.unselectedBackgroundColor;
}

- (void)setUnselectedBackgroundColor:(UIColor *)unselectedBackgroundColor {
    self.selectionCustomization.unselectedBackgroundColor = unselectedBackgroundColor;
}

- (UIColor *)unselectedBorderColor {
    return self.selectionCustomization.unselectedBorderColor;
}

- (void)setUnselectedBorderColor:(UIColor *)unselectedBorderColor {
    self.selectionCustomization.unselectedBorderColor = unselectedBorderColor;
}

@end

File: StripePayments/StripePayments/PaymentHandler/STPThreeDSSelectionCustomization.swift
//
//  STPThreeDSSelectionCustomization.swift
//  StripePayments
//
//  Created by Yuki Tokuhiro on 6/18/19.
//  Copyright © 2019 Stripe, Inc. All rights reserved.
//

import Foundation
import UIKit

#if canImport(Stripe3DS2)
    import Stripe3DS2
#endif

/// A customization object that configures the appearance of
/// radio buttons and checkboxes.
public class STPThreeDSSelectionCustomization: NSObject {
    /// The default settings.
    @objc
    public class func defaultSettings() -> STPThreeDSSelectionCustomization {
        return STPThreeDSSelectionCustomization()
    }

    internal var selectionCustomization = STDSSelectionCustomization.defaultSettings()

    /// The primary color of the selected state.
    /// Defaults to blue.

    @objc public var primarySelectedColor: UIColor {
        get {
            return selectionCustomization.primarySelectedColor
        }
        set(primarySelectedColor) {
            selectionCustomization.primarySelectedColor = primarySelectedColor
        }
    }
    /// The secondary color of the selected state (e.g. the checkmark color).
    /// Defaults to white.

    @objc public var secondarySelectedColor: UIColor {
        get {
            return selectionCustomization.secondarySelectedColor
        }
        set(secondarySelectedColor) {
            selectionCustomization.secondarySelectedColor = secondarySelectedColor
        }
    }
    /// The background color displayed in the unselected state.
    /// Defaults to light blue.

    @objc public var unselectedBackgroundColor: UIColor {
        get {
            return selectionCustomization.unselectedBackgroundColor
        }
        set(unselectedBackgroundColor) {
            selectionCustomization.unselectedBackgroundColor = unselectedBackgroundColor
        }
    }
    /// The color of the border drawn around the view in the unselected state.
    /// Defaults to blue.

    @objc public var unselectedBorderColor: UIColor {
        get {
            return selectionCustomization.unselectedBorderColor
        }
        set(unselectedBorderColor) {
            selectionCustomization.unselectedBorderColor = unselectedBorderColor
        }
    }

}
